interoptest/src/goservice/testservice/request_processor.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testservice
+
+import (
+	"fmt"
+	"github.com/census-ecosystem/opencensus-experiments/interoptest/src/goservice/genproto"
+	"golang.org/x/net/context"
+	"sync"
+)
+
+// RequestProcessor is the type that process test requests.
+type RequestProcessor struct {
+	httpSender *HttpSender
+	grpcSender *GRPCSender
+}
+
+var instance *RequestProcessor
+var once sync.Once
+
+func (rp *RequestProcessor) getInstance() *RequestProcessor {
+	once.Do(func() {
+		instance = &RequestProcessor{httpSender: NewHttpSender(), grpcSender: NewGRPCSender()}
+	})
+	return instance
+}
+
+func (rp *RequestProcessor) send(ctx context.Context, serviceHop interop.ServiceHop, req *interop.TestRequest) (*interop.TestResponse, error) {
+	if serviceHop.GetService().GetSpec().GetTransport() == interop.Spec_GRPC {
+		return rp.grpcSender.Send(ctx, serviceHop, req)
+	} else if serviceHop.GetService().GetSpec().GetTransport() == interop.Spec_HTTP {
+		return rp.httpSender.Send(ctx, serviceHop, req)
+	}
+	return nil, fmt.Errorf("invalid transport")
```

Comment:
nit: you can make this error a global var to avoid re-creating it.

Reply:
ok